src/teehr/evaluation/tables/base_table.py

```diff
@@ -156,13 +162,25 @@ def _validate(self, df: ps.DataFrame, strict: bool = True) -> ps.DataFrame:
             If True, any extra columns will be dropped before validation.
             If False, will be validated as-is.
             The default is True.
+
+        Returns
+        -------
+        validated_df : ps.DataFrame
+            The validated DataFrame.
         """
         schema = self._get_schema()
 
         logger.info(f"Validating DataFrame with {schema.columns}.")
 
+        schema_cols = schema.columns.keys()
+
+        # Add missing columns
+        if add_missing_columns:
+            for col_name in schema_cols:
+                if col_name not in df.columns:
+                    df = df.withColumn(col_name, lit(None))
```

Comment:
just to make sure I understand, here we're adding an empty column(s) (column names that exist in the schema but not dataframe), then in line 186 it coerces the empty column(s) to the correct data type as defined by the schema?